src/ekore/harmonics/cache.py

```diff
@@ -0,0 +1,236 @@
+"""Caching harmonic sums across :mod:`ekore`."""
+import numba as nb
+import numpy as np
+import numpy.typing as npt
+
+from . import w1, w2, w3, w4, w5
+from .g_functions import mellin_g3
+from .polygamma import recursive_harmonic_sum
+
+# here a register of all possible functions
+S1 = 0  # = S_1(N)
+S2 = 1
+S3 = 2
+S4 = 3
+S5 = 4
+Sm1 = 5
```

Comment:
If you wish, a slightly more flexible option might be:
```suggestion
_index = iter(range(1000))
S1 = next(_index)  # = S_1(N)
S2 = next(_index)
S3 = next(_index)
S4 = next(_index)
S5 = next(_index)
Sm1 = next(_index)
```

Because you do not really care about the actual number, for as long as it is a stable one, and in this way they would be assigned at import time (not from literals, but still fully constants).
This would also make it simpler to insert in the middle, e.g.
```py
_index = iter(range(1000))
S1 = next(_index)  # = S_1(N)
S2 = next(_index)
S3 = next(_index)
S4 = next(_index)
S5 = next(_index)
S6 = next(_index)
Sm1 = next(_index)
```
modifying only the line involved.
Furthermore, it would stress that the actual number is irrelevant.

However, they are minor points, so it is definitely a detail.

Reply:
The number in the iterator is just a stupidly high number.
You don't care about the exact one, if available it could have been even infinite, since if you're not iterating until infinite you simply use the initial segment, whatever the length is.

Reply:
thanks for this suggestion, we can indeed do this way

Reply:
done in https://github.com/NNPDF/eko/pull/253/commits/cd34b1c666755450c64c19ec49a463a2f959266b